tests/testthat/test-init.R

```diff
@@ -46,3 +46,13 @@ test_that("site meta doesn't break unexpectedly", {
 
   expect_snapshot(yaml)
 })
+
+test_that("override works correctly for as_pkgdown", {
```

Comment:
I'd suggest putting this test in `test-package.R` since that's corresponds to the file where you changed the code.